BingWallpaper/Downloader.cs

```diff
@@ -28,13 +28,25 @@ public Downloader(IWebAccessor webAccessor)
             State = LoadState();
         }
 
-        public void Run()
+        public async void Run(int refrashTime)
         {
-            if (_task != null)
+            if (refrashTime > 0)
             {
-                throw new InvalidOperationException("Already started running");
+                if (_task != null && !_task.IsCompleted)
+                {
+                    throw new InvalidOperationException("Already started running");
+                }
+                while (!_cts.Token.IsCancellationRequested)
+                {
+                    _task = DoWork(_cts.Token);
+                    _firstTime = false;
+                    await Task.Delay(TimeSpan.FromMinutes(refrashTime), _cts.Token);
+                }
+            }
+            else
+            {
+                _task = DoWork(_cts.Token);
```

Comment:
When `refreshTime` is 0, I think we should cancel the current work before starting a new work.